api/src/main/java/jakarta/enterprise/lang/model/declarations/ParameterInfo.java

```diff
@@ -2,12 +2,40 @@
 
 import jakarta.enterprise.lang.model.types.Type;
 
+/**
+ * Provides read-only information about a method parameter.
+ */
 public interface ParameterInfo extends DeclarationInfo {
-    String name(); // TODO doesn't have to be known
+    /**
+     * Returns the zero-based position of this parameter in the method declaration.
+     *
+     * @return the position of this parameter in the method declaration
+     */
+    // TODO one-based?
+    int position();
```

Comment:
Not entirely sure if this can be implemented with annotation processor?

Reply:
Yes it can be, but better to be zero based no?

Reply:
I don't really know if this is necessary, I just thought I'd propose it and see was you think. I'd be fine with removing -- especially if we do a consistent naming strategy for parameters with unknown names (which you suggest below and I agree).

Reply:
I removed this, because I think with predictable `name`, there's no need for it.